modules/ingest-user-agent/src/main/java/org/elasticsearch/ingest/useragent/IngestUserAgentPlugin.java

```diff
@@ -53,9 +54,15 @@
 
     static Map<String, UserAgentParser> createUserAgentParsers(Path userAgentConfigDirectory, UserAgentCache cache) throws IOException {
         Map<String, UserAgentParser> userAgentParsers = new HashMap<>();
+        InputStream deviceTypeRegexStream = IngestUserAgentPlugin.class.getResourceAsStream("/device_type_regexes.yml");
+
+        if (Files.exists(PathUtils.get(String.valueOf(userAgentConfigDirectory), "/device_type_regexes.yml")) == false) {
+            deviceTypeRegexStream = null;
+        }
```

Comment:
Is this check necessary if `device_type_regexes.yml` is a resource within the JAR file?